Logging/LogHandler/FileLogHandler.cs

```diff
@@ -118,21 +132,18 @@ public FileLogHandler(string path, LogType logType = LogType.Append, bool should
 
             CreateWriter();
 
-            if (logType == LogType.Replace && shouldRotate)
+            if (m_WriteMode == WriteMode.Replace && m_RotateFileSizeLimit.HasValue)
             {
-                int index = 1;
-                string rotatedFilePath = GetRotatedFilePath(index);
-                while (File.Exists(rotatedFilePath))
+                foreach ((_, string filePath) in GetRotatedFiles())
                 {
-                    File.Delete(rotatedFilePath);
-                    rotatedFilePath = GetRotatedFilePath(++index);
+                    File.Delete(filePath);
```

Comment:
Should we log what is being deleted?

Reply:
I don't think so, if you've explicitly chosen `WriteMode: Replace` then you expect the files (rotated or not) to be replaced. We didn't send a log pre-rotation when replacing an existing log.